### PR TITLE
fix: moved registration tests to specific files.

### PR DESCRIPTION
**Pull request by attiyaIshaque**
### Description

All tests that are not directly related to the registration form functionality have been moved to their respective component-specific test files. 

#### JIRA

[VAN-1628](https://2u-internal.atlassian.net/browse/VAN-1628)

### CODE REVIEW

File: src/register/RegistrationPage.test.jsx [hunk 1]
```diff
@@ -153,11 +148,7 @@ describe('RegistrationPage', () => {
   };
 
   const ssoProvider = {
```
Comment (mubbsharanwar) [inline src/register/RegistrationPage.test.jsx:150]: Do we still need to pass ssoProvider in `formPayload`? Reference line 216.

File: src/register/components/RegistrationFailure.test.jsx [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+  };
+
+  beforeEach(() => {
+    store = mockStore(initialState);
+    configure({
+      loggingService: { logError: jest.fn() },
+      config: {
+        ENVIRONMENT: 'production',
+        LANGUAGE_PREFERENCE_COOKIE_NAME: 'yum',
+      },
```
Comment (mubbsharanwar) [inline src/register/components/RegistrationFailure.test.jsx:116]: Do we need these configs?

File: src/register/components/RegistrationFailure.test.jsx [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('Test Registration Failure', () => {
+    mergeConfig({
+      SHOW_CONFIGURABLE_EDX_FIELDS: true,
+    });
```
Comment (mubbsharanwar) [inline src/register/components/RegistrationFailure.test.jsx:133]: Also what about these?

Comment (attiyaIshaque) [reply src/register/components/RegistrationFailure.test.jsx]: I have removed these.

File: src/register/components/RegistrationFailure.test.jsx (hunk 2 above)
Comment (attiyaIshaque) [reply src/register/components/RegistrationFailure.test.jsx:116]: Yes. we can Pass ENVIRONMENT undefined.

File: src/register/RegistrationPage.test.jsx (hunk 1 above)
Comment (attiyaIshaque) [reply src/register/RegistrationPage.test.jsx]: Nope, we don't need this. Thanks for pointing this out.

**Pull request merged**